docs/investigation/analytics-collection.md

```diff
@@ -0,0 +1,100 @@
+# Analytics collection
+
+This document contains a summary of a short research around collecting anonymous analytics (a.k.a. telemetry) for BotKube.
+
+## Goal
+
+The main goal is learning how BotKube is used, in the most private and least invasive way for users.
+All data should be anonymized to prevent identifying users. All analytics will be used only as aggregated collection of data (in forms of some statistics), to further improve BotKube and adjust its roadmap.
+
+The change should be widely communicated to all users, and there should be an ability to opt out from collecting such analytics.
+
+## How others do it
+
+I tried to look for some software from CNCF landscape which has telemetry built-in. Unfortunately, I couldn't many examples when it comes to this field. It was much easier to find something in JavaScript world (e.g. NextJS).
+
+Anyway, here are a few examples I found:
+
+- [`odo`](https://github.com/redhat-developer/odo) - Twillio Segment ([source](https://github.com/redhat-developer/odo/blob/77d6b6df5cdd05074db8728d1aead76d3e259e25/pkg/segment/segment.go))
+- [`config-syncer` (previously `kubed`)](https://github.com/kubeops/config-syncer) - Google Analytics ([source](https://github.com/kubeops/config-syncer/blob/release-0.12/vendor/kmodules.xyz/client-go/tools/cli/cli.go))
+- [`netdata`](https://github.com/netdata/netdata) - PostHog ([see docs](https://learn.netdata.cloud/docs/agent/anonymous-statistics))
+- [Testkube](https://github.com/kubeshop/testkube) - Google Analytics 4 ([source](https://github.com/kubeshop/testkube/blob/34c57dbdb9312b68910e0ad5808485292fa31313/pkg/analytics/analytics.go))
+- Rancher backed off from this idea when it comes to k3s: [see issue](https://github.com/k3s-io/k3s/issues/834)
+
+All of the software I found that collect anonymous analytics (not only from Go/Kubernetes space) have opt-out based configuration.
+
+## Initial data to collect
+
+This list contains initial data which we could collect.
+
+- Running installations
```

Comment:
🤔 Do you mean notifying Segment every time we got an event? This could be resource-expensive when receiving multiple events. We would need to think about batching them somehow 🤔  

EDIT: Sounds like it is even built-in to the Segment library - https://segment.com/docs/connections/sources/catalog/libraries/server/go/#batching. Adding it to the list then 👍